src/async_fuse/memfs/s3_metadata.rs

```diff
@@ -68,24 +72,29 @@ pub struct S3MetaData<S: S3BackEnd + Send + Sync + 'static> {
 }
 
 /// Parse S3 info
-fn parse_s3_info(info: &str) -> (&str, &str, &str, &str) {
+fn parse_s3_info(info: &str) -> DatenLordResult<(&str, &str, &str, &str)> {
     info.split(S3_INFO_DELIMITER)
         .next_tuple()
-        .unwrap_or_else(|| panic!("parse s3 information failed. s3_info: {info}"))
+        .ok_or_else(|| anyhow::anyhow!("parse s3 information failed. s3_info: {info}").into())
```

Comment:
It's a very general input parsing error, we can add an error type for it, or maybe we've already have one.

Reply:
We do have one, called `S3Error`, but it's not designed well. We can refactor it in the future.